src/common/reducers/repositories.js

```diff
@@ -2,10 +2,11 @@ import * as ActionTypes from '../actions/repositories';
 import { parseGitHubRepoUrl } from '../helpers/github-url';
 
 export function repositories(state = {
-  isFetching: false,
+  // has no server side content, so initial state is always 'fetching'
+  isFetching: true,
```

Comment:
I'm changing this back because it's not relevant to the PR, but I'm not completely convinced by your arguments (neither convinced my own solution) :)

The reason I'm not convinced by the argument is we don't set fetching state when a component is fetching, we set it before a component is fetching, so the true meaning of `isFetching` isn't quite right in either case (some failure could prevent an actual fetch in both cases) ... but from a user's point of view, from the moment that HTML is rendered, that is *before* React has reconciled the tree, that component is in a state of "fetching" ... fetching js, fetching data, whatever, but because there's a fetch API it sounds more like we mean we're engaged in request fetch.